contracts/src/emergency/sequencer_uptime_feed.cairo

```diff
@@ -0,0 +1,358 @@
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 0_u8),
+                value.answer.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 1_u8),
+                value.block_num.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 2_u8),
+                value.started_at.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 3_u8),
+                value.updated_at.into()
+            )
+        }
+    }
+
+
+    struct Storage {
+        // l1 sender is an ethereum address
+        _l1_sender: u256,
+        // maps round id to round transmission
+        _round_transmissions: LegacyMap<u128, RoundTransmission>,
+        _latest_round_id: u128,
+    }
+
+    #[event]
+    fn RoundUpdated(status: bool, updated_at: u64) {}
+
+    #[event]
+    fn NewRound(round_id: u128, started_by: ContractAddress, started_at: u64) {}
+
+    #[event]
+    fn AnswerUpdated(current: bool, round_id: u128, timestamp: u64) {}
+
+    #[event]
+    fn UpdateIgnored(latest_status: bool, latest_timestamp: u64, incoming_status: bool, incoming_timestamp: u64) {}
+
+    #[event]
+    fn L1SenderTransferred(from_address: u256, to_address: u256) {}
+
+    impl Aggregator of IAggregator {
+        fn latest_round_data() -> Round {
+            require_access();
+            let latest_round_id = _latest_round_id::read();
+            let round_transmission = _round_transmissions::read(latest_round_id);
+            // bool -> felt252 -> u128
+            let answer: u128 = round_transmission.answer.into().try_into().unwrap();
+            Round {
+                round_id: latest_round_id.into(),
+                answer: answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.started_at,
+                updated_at: round_transmission.updated_at,   
+            }
+        }
+
+        fn round_data(round_id: u128) -> Round {
+            require_access();
+            assert(round_id < _latest_round_id::read(), 'invalid round id');
+            let round_transmission = _round_transmissions::read(round_id);
+            // bool -> felt252 -> u128
+            let answer: u128 = round_transmission.answer.into().try_into().unwrap();
+            Round {
+                round_id: round_id.into(),
+                answer: answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.started_at,
+                updated_at: round_transmission.updated_at,   
+            }
+        }
+
+        fn description() -> felt252 {
+            'L2 Sequencer Uptime Status Feed'
```

Comment:
No need to create storage variables (expensive) for these values which won't be modified